services/src/main/scala/cromwell/pipeline/service/ProjectVersioning.scala

```diff
@@ -1,58 +1,91 @@
 package cromwell.pipeline.service
 
-import cromwell.pipeline.datastorage.dto._
 import java.nio.file.Path
-import scala.concurrent.{ ExecutionContext, Future }
 
-trait ProjectVersioning[E >: VersioningException] {
-  type AsyncResult[T] = Future[Either[E, T]]
-  type ProjectFiles = List[ProjectFile]
-
-  def updateFile(project: Project, projectFile: ProjectFile, version: PipelineVersion)(
-    implicit ec: ExecutionContext
-  ): AsyncResult[UpdateFiledResponse]
+import cromwell.pipeline.datastorage.dto._
 
-  def updateFiles(project: Project, projectFiles: ProjectFiles)(
-    implicit ec: ExecutionContext
-  ): AsyncResult[List[SuccessResponseMessage]]
+import scala.concurrent.Future
 
-  def createRepository(localProject: LocalProject)(
-    implicit ec: ExecutionContext
-  ): AsyncResult[Project]
+trait ProjectVersioning[E <: VersioningException] {
+  type AsyncResult[T] = Future[Either[E, T]]
+  type ProjectFiles = List[ProjectFile]
 
-  def getFiles(project: Project, version: Option[PipelineVersion] = None)(
-    implicit ec: ExecutionContext
-  ): AsyncResult[List[ProjectFile]]
+  /**
+   * Creates versioned repository for local project
+   * @param localProject project that exists only locally and does not have real repository
+   * @return full project with versioning-specific fields (`repositoryId`, `version`)
+   */
+  def createRepository(localProject: LocalProject): AsyncResult[Project]
 
-  def getProjectVersions(project: Project)(
-    implicit ec: ExecutionContext
-  ): AsyncResult[Seq[GitLabVersion]]
+  /**
+   * Update files in repository with versioning
+   * @param project project to which file belongs to
+   * @param projectFile files to update
+   * @param version preferable project version
+   * @return new project version
+   */
+  def updateFile(
+    project: Project,
+    projectFile: ProjectFile,
+    version: Option[PipelineVersion]
+  ): AsyncResult[PipelineVersion]
 
-  def getFileCommits(project: Project, path: Path)(
-    implicit ec: ExecutionContext
-  ): AsyncResult[Seq[FileCommit]]
+  /**
+   * Update files in repository with versioning
+   * @param project project to which files belong to
+   * @param projectFiles files to update
+   * @param version preferable project version
+   * @return new project version
+   */
+  def updateFiles(
+    project: Project,
+    projectFiles: ProjectFiles,
+    version: Option[PipelineVersion]
+  ): AsyncResult[PipelineVersion]
 
-  def getFileVersions(project: Project, path: Path)(
-    implicit ec: ExecutionContext
-  ): AsyncResult[Seq[GitLabVersion]]
+  /**
+   * Retrieves file content from repository
```

Comment:
Since `ProjectFileContent` is a parameter of `ProjectFile` and the method returns a whole ProjectFile, this may be confusing

Reply:
Chaned to `Retrieves project file from repository`